src/manifest.c

```diff
@@ -1116,6 +1116,34 @@ struct list *consolidate_files(struct list *files)
 	return list;
 }
 
+/*
+ * Takes a list of files, and removes files from the list that exist on the filesystem.
+ * This is used for cases like bundle-add where we don't want to re-add files that
+ * already are on the system.
+ */
+struct list *filter_out_existing_files(struct list *files)
+{
+	struct list *list, *next, *ret;
+	struct file *file;
+
+	ret = files;
+
+	list = list_head(files);
+	while (list) {
+		next = list->next;
+
+		file = list->data;
+
+		if (verify_file_lazy(file->filename)) {
```

Comment:
This needs to perform the check under `path_prefix`. This is one reason so many tests are failing.
```C
if (verify_file_lazy(mk_full_filename(path_prefix, file->filename))) {
```